src/cmd/edit.rs

```diff
@@ -0,0 +1,267 @@
+use crate::cmd::{Edit, Run};
+use crate::db::{Database, DatabaseFile, Epoch, Rank};
+use crate::util::resolve_path;
+use crate::{config, util};
+use anyhow::Result;
+use std::fmt::Write as FmtWrite;
+use std::path::PathBuf;
+
+use dialoguer::Input;
+use edit::edit;
+
+const HEADER: &str = "\
+# Blank lines and lines prepended with '#' are ignored; Line order is insignificant
+# last_accessed,rank,path
+";
+
+enum ValidationResult {
+    Success,
+    Retry,
+    Exit,
+}
+
+struct Problem {
+    line_number: usize,
+    text: String,
+}
+
+struct Problems {
+    warnings: Vec<Problem>,
+    errors: Vec<Problem>,
+}
+
+impl Problem {
+    fn new(line_number: usize, text: String) -> Self {
+        Problem { line_number, text }
+    }
+}
+
+impl Run for Edit {
+    fn run(&self) -> Result<()> {
+        while let Some(db_edits) = get_db_edits()? {
+            let data_dir = config::data_dir()?;
+            let mut db_file = DatabaseFile::new(data_dir);
+            let mut db = db_file.open()?;
+            db.clear();
+            let problems = get_problems(&mut db, db_edits);
+            let result = handle_problems(problems);
+            match result {
+                ValidationResult::Success => {
+                    db.save()?;
+                    return Ok(());
+                }
+                ValidationResult::Exit => break,
+                ValidationResult::Retry => continue,
+            }
+        }
+        println!("Zoxide database not altered");
+        Ok(())
+    }
+}
+
+fn get_db_edits() -> Result<Option<String>> {
+    let data_dir = config::data_dir()?;
+    let mut db = DatabaseFile::new(data_dir);
+    let mut db = db.open()?;
+    let mut stream = db.stream(util::current_time().unwrap());
+    let mut to_edit = String::from(HEADER);
+    while let Some(dir) = stream.next() {
+        writeln!(&mut to_edit, "{},{},{}", dir.last_accessed, dir.rank, dir.path)?;
```

Comment:
Rank is an `f64`. That's potentially a lot of decimals that the user doesn't really care about. We could just print the first 2 decimal places here and it should be fine.